spec/requests/workflow_instances_spec.rb

```diff
@@ -0,0 +1,66 @@
+describe "WorkflowInstances API" do
+  context "GET /api/workflow_instances" do
+    context "without an appropriate role" do
+      before { api_basic_authorize }
+
+      it "returns forbidden" do
+        FactoryBot.create(:workflow_instance)
+
+        get(api_workflow_instances_url)
+
+        expect(response).to have_http_status(:forbidden)
+      end
+    end
+
+    context "with appropriate role" do
+      before { api_basic_authorize(collection_action_identifier(:workflow_instances, :read, :get)) }
+
+      it "returns all Workflow Instances" do
+        workflow_instance = FactoryBot.create(:workflow_instance)
+
+        get(api_workflow_instances_url)
+
+        expected = {
+          "name"      => "workflow_instances",
+          "resources" => [{"href" => api_workflow_instance_url(nil, workflow_instance)}]
+        }
+        expect(response).to have_http_status(:ok)
+        expect(response.parsed_body).to include(expected)
+      end
+    end
+  end
+
+  context "GET /api/workflow_instances/:id" do
+    context "without an appropriate role" do
+      before { api_basic_authorize }
+
+      it "returns forbidden" do
+        workflow_instance = FactoryBot.create(:workflow_instance)
+
+        get(api_workflow_instance_url(nil, workflow_instance))
+
+        expect(response).to have_http_status(:forbidden)
+      end
+    end
+
+    context "with appropriate role" do
+      before { api_basic_authorize(action_identifier(:workflow_instances, :read, :resource_actions, :get)) }
+
+      it "returns a single Workflow" do
+        workflow_instance = FactoryBot.create(:workflow_instance)
+
+        get(api_workflow_instance_url(nil, workflow_instance))
+
+        expected = {
+          "id"               => workflow_instance.id.to_s,
+          "context"          => workflow_instance.context,
+          "credentials"      => workflow_instance.credentials,
```

Comment:
Maybe for a follow up pr, but might be nice to show that this does _not_ expose the password

Reply:
Possibly, I think we want to keep actual secrets out of this payload entirely though so this would only have the JSON path reference. It would be helpful to see what credential value it is using is.
Maybe if there is no JSON path we blank out the value in case they put a value directly in there or we prevent non-references from even being added